openedx/features/course_experience/views/course_updates.py

```diff
@@ -48,17 +50,34 @@ def render_to_fragment(self, request, course_id=None, **kwargs):
         course_url_name = default_course_url_name(request)
         course_url = reverse(course_url_name, kwargs={'course_id': unicode(course.id)})
 
-        # Fetch the updates as HTML
-        updates_html = get_course_info_section(request, request.user, course, 'updates')
+        # Fetch all of the updates individually
+        info_module = get_course_info_section_module(request, request.user, course, 'updates')
+        ordered_updates = self.order_updates(info_module.items)
+
+        # Older implementations store a single html object representing all the updates
+        plain_html_updates = info_module.data
 
         # Render the course home fragment
         context = {
             'csrf': csrf(request)['csrf_token'],
             'course': course,
             'course_url': course_url,
-            'updates_html': updates_html,
+            'updates': ordered_updates,
+            'update_list_html': plain_html_updates,
             'disable_courseware_js': True,
             'uses_pattern_library': True,
         }
         html = render_to_string('course_experience/course-updates-fragment.html', context)
         return Fragment(html)
+
+    @staticmethod
+    def order_updates(updates):
```

Comment:
Can you remove the other ``ordered_updates`` method and have its callers call this method instead. That way this logic isn't duplicated.